### PR TITLE
Array returned from Java object not converted to JavaScript array #5501

### PR DESCRIPTION
**Pull request by anatol-sialitski**

### CODE REVIEW

File: modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java [hunk 1]
```diff
@@ -32,6 +32,11 @@ public Object toJs( final Object value )
             return toJs( (List) value );
         }
 
+        if ( value != null && value.getClass().isArray() )
+        {
+            return toJs( (Object[]) value );
```
Comment (rymsha) [inline modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java:37]: you could use Arrays.asList

**Review by rymsha — APPROVED** [review]

**Review by vbradnitski — APPROVED** [review]

File: modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java [hunk 2]
```diff
@@ -32,6 +33,11 @@ public Object toJs( final Object value )
             return toJs( (List) value );
         }
 
+        if ( value != null && value.getClass().isArray() )
+        {
+            return toJs( Arrays.asList( (Object[]) value ) );
```
Comment (rymsha) [inline modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java:38]: Not all arrays can be casted to Object[]

**Review by rymsha — CHANGES_REQUESTED** [review]

File: modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java (hunk 2 above)
Comment (anatol-sialitski) [reply modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java]: done

File: modules/script/script-impl/src/test/java/com/enonic/xp/script/impl/util/JsObjectConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        assertTrue( result instanceof ScriptObjectMirror );
+        assertTrue( ( (ScriptObjectMirror) result ).isArray() );
+        assertEquals( 3, ( (ScriptObjectMirror) result ).size() );
+    }
+
+    @Test
+    public void testToJs_Primitives()
+    {
+        final int[] stringValues = {1, 2, 3};
```
Comment (rymsha) [inline modules/script/script-impl/src/test/java/com/enonic/xp/script/impl/util/JsObjectConverterTest.java:44]: int values?

File: modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java [hunk 4]
```diff
@@ -143,4 +150,23 @@ else if ( NashornHelper.isDateType( source ) )
     {
         return arg -> toObject( source.call( null, arg ) );
     }
+
+    private Object[] createObjectArray( final Object objectArray )
+    {
+        if ( !objectArray.getClass().getComponentType().isPrimitive() )
```
Comment (rymsha) [inline modules/script/script-impl/src/main/java/com/enonic/xp/script/impl/util/JsObjectConverter.java:156]: probably better to leave primitive arrays as is.

File: modules/script/script-impl/src/test/java/com/enonic/xp/script/impl/util/JsObjectConverterTest.java (hunk 3 above)
Comment (anatol-sialitski) [reply modules/script/script-impl/src/test/java/com/enonic/xp/script/impl/util/JsObjectConverterTest.java]: done

**Review by rymsha — APPROVED** [review]

**Pull request merged**